.github/workflows/release-internal.yml

```diff
@@ -32,20 +32,55 @@ on:
         required: true
         default: false
 
+  schedule:
+    - cron: '0 13 * * 3' # Every Wednesday at 8:00 AM central time
+
 defaults:
   run:
     shell: bash
 
 jobs:
+  get-last-successful-release-run:
+    runs-on: ubuntu-latest
+    outputs:
+      commit_sha: ${{ steps.get_last_successful_release_run.outputs.commit_sha }}
+    steps:
+      - name: Get last successful Release to Cloud run details
+        id: get_last_successful_release_run
+        run: |
+          last_run_url=$(curl --silent -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/${{ github.repository }}/actions/runs?event=workflow_dispatch&status=success&branch=main&workflow_file=release-internal.yml" | jq -r '.workflow_runs[0].url')
+          echo "Last run URL: $last_run_url"
+          commit_sha=$(curl --silent -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" $last_run_url | jq -r '.head_sha')
```

Comment:
Is there a way this can be `None`? Might be nice to validate and exit here if it's empty.